cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -0,0 +1,165 @@
+#include "refactor.hpp"
+
+#include <mg_utils.hpp>
+#include <unordered_set>
+
+void Refactor::InsertCloneNodesRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const int cycle_id,
+                                      const int node_id) {
+  auto *record = mgp::result_new_record(result);
+
+  mg_utility::InsertIntValueResult(record, std::string(kResultClonedNodeId).c_str(), cycle_id, memory);
+  mg_utility::InsertNodeValueResult(graph, record, std::string(kResultNewNode).c_str(), node_id, memory);
+}
+
+mgp::Node GetStandinOrCopy(mgp::List &standin_nodes, mgp::Node node,
+                           std::map<mgp::Node, mgp::Node> &old_new_node_mirror) {
+  for (auto pair : standin_nodes) {
+    if (!pair.IsList() || !pair.ValueList()[0].IsNode() || !pair.ValueList()[1].IsNode()) {
+      throw mgp::ValueException(
+          "Configuration map must consist of specific keys and values described in documentation.");
+    }
+    if (node == pair.ValueList()[0].ValueNode()) {
+      return pair.ValueList()[1].ValueNode();
+    }
+  }
+  try {
+    return old_new_node_mirror.at(node);  // what if they send me wrong path
+  } catch (const std::out_of_range &e) {
+    throw mgp::ValueException("Can't clone relationship without cloning relationship's source and/or target nodes.");
+  }
+}
+
+bool CheckIfStandin(mgp::Node node, mgp::List standin_nodes) {
```

Comment:
You can pass these params here as const ref. Here we don't need to copy them, although it is a cheap copy. So correct params passing would be:
`bool CheckIfStandin(const mgp::Node &node, const mgp::List &standin_nodes)`